src/track/track.cpp

```diff
@@ -1062,6 +1062,23 @@ CuePointer Track::findCueById(DbId id) const {
     return CuePointer();
 }
 
+CuePointer Track::findHotcueByIndex(int idx) const {
+    auto locked = lockMutex(&m_qMutex);
+    auto is_hotcue_n = [](int n) {
+        return [n](const CuePointer& pCue) {
+            return pCue && pCue->getHotCue() == n;
+        };
+    };
+    auto cueIt = std::find_if(m_cuePoints.begin(), m_cuePoints.end(), is_hotcue_n(idx));
```

Comment:
any specific reason for the double lambda? It could also have just been one.

Reply:
Hehe, I didn't question that when I accepted you're proposal.

I have no clue how to capture the iterator and n with that function.

But this seems to work:
```suggestion
    auto cueIt = std::find_if(
            m_cuePoints.begin(),
            m_cuePoints.end(),
            [&idx](const auto pCue) {
                return pCue && pCue->getHotCue() == idx;
            });
```

Reply:
I did that? Yeah seems like the overcomplicated stuff I like to do :sweat_smile: 

Probably just wanted to give it a name. I should really try to KISS more often.

What you posted works but is suboptimal because 

1. you're capturing the `int idx` by reference (just capture it by value as per the usual guidance on cheap to copy types).
2. you're taking the pCue by value (`auto` never deduces to a reference, See scott meyers "Effective Modern C++" for a very good explanation on type deduction). Since pCue is some sort of reference-counted shared pointer this causes unnecessary changes in the reference count. So instead take a reference to the pointer like in the original lambda.

We can also keep it like this, just wanted to give my two cents on your proposal.